src/test/scala/reporter/CloudWatchReporterSpec.scala

```diff
@@ -0,0 +1,35 @@
+package reporter
+
+import com.blacklocus.metrics.CloudWatchReporterBuilder
+import com.codahale.metrics.{Metric, MetricFilter, MetricRegistry}
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{FlatSpec, Matchers, PrivateMethodTester}
+import Utils.buildMetricFilter
+import scala.util.matching.Regex
+
+
+class CloudWatchReporterSpec extends FlatSpec with Matchers with PrivateMethodTester with MockFactory {
+
+  private val metricRegistry: MetricRegistry = new MetricRegistry
+  private val metric: Metric = mock[Metric]
+
+
+  "Metrics filter" should "match any metric when empty string filter is given" in {
+    val filter = buildMetricFilter("")
+
+    filter.matches("any_metrics_name", metric) should be(true)
+    filter.matches("xfaewojz", metric) should be(true)
+  }
+
+  "Metrics filter" should "match metrics that have the regex pattern: true" in {
```

Comment:
Nitpick: you can just use `it` instead of `"Metrics filter"` for tests after the first one